python/pyarrow/ipc.pxi

```diff
@@ -900,6 +925,31 @@ cdef class _RecordBatchFileReader(_Weakrefable):
     # time has passed
     get_record_batch = get_batch
 
+    def get_batch_with_custom_metadata(self, int i):
+        """
+        Read the record batch with the given index along with its custom metadata
+
+        Parameters
+        ----------
+        i : int
+            The index of the record batch in the IPC file.
+
+        Returns
+        -------
+        batch : RecordBatch
+        custom_metadata: dict
```

Comment:
@jorisvandenbossche Is this the right Numpydoc syntax when a namedtuple is returned?